src/CMakeLists.txt

```diff
@@ -14,10 +14,21 @@ include(CIME_utils)
 set(CLM_ROOT "..")
 
 # find needed external packages
-# This is where ESMF could be asked for, but it's already included in the share build brought in below
 # NetCDF is required -- because PIO and NetCDF are required by the standard default ESMF libraries
 find_package(NetCDF 4.7.4 REQUIRED Fortran)
 
+# The following - for finding ESMF - is copied from the share CMakeLists.txt
```

Comment:
Excellent, I was expecting to need to have this when I initially got this working on Derecho. It always seemed to conflict with elsewhere were ESMF was defined. And I couldn't get it to work with something like this there. In that PR none of us where perfectly satisfied, but since it worked, we went with it. I'm glad to see the proper treatment in place now.

And by the way, this fixes #2452 which is awesome!

Reply:
I also think that we should state that we expect PIO to be pre-built into the ESMF version used here. I mention this below. But, this might need a discussion.